lib/screens/surveydetails/widget/question_paging_widget.dart

```diff
@@ -0,0 +1,203 @@
+                ),
+                child: Icon(
+                  Icons.close,
+                  color: Theme.of(context).primaryColor,
+                ),
+              )
+            : const SizedBox(),
+      ],
+    );
+  }
+
+  Widget _buildQuestionHeader(BuildContext context, QuestionUiModel question) {
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: [
+        if (!_isIntroQuestion(question)) ...[
+          Text(
+            AppLocalizations.of(context)!.question_number(
+              _currentIndex,
+              widget.questions.length - 1,
+            ),
+            style:
+                Theme.of(context).textTheme.titleLarge?.copyWith(fontSize: 15),
+          ),
+          const SizedBox(height: Dimensions.paddingSmall),
+          Text(
+            question.text,
+            style: Theme.of(context).textTheme.titleLarge,
+          ),
+        ] else ...[
+          Text(
+            question.title,
+            style: Theme.of(context).textTheme.titleLarge,
+          ),
+          Padding(
+            padding: const EdgeInsets.only(top: 16),
+            child: Text(
+              question.text,
+              style: Theme.of(context).textTheme.bodyMedium,
+            ),
+          ),
+        ],
+      ],
+    );
+  }
+
+  Widget _buildQuestionFooter(
+    BuildContext context,
+    QuestionUiModel question,
+    PageController controller,
+  ) {
+    return Row(
+      mainAxisAlignment: MainAxisAlignment.end,
+      children: [
+        Padding(
+            padding: const EdgeInsets.only(left: 20),
+            child: _isIntroQuestion(question)
+                ? _startSurveyButtonWidget(context, controller)
+                : _isLastQuestion(question)
+                    ? _submitSurveyButtonWidget(context)
+                    : _nextButtonWidget(context, controller)),
+      ],
+    );
+  }
+
+  Widget _nextButtonWidget(BuildContext context, PageController controller) {
+    return FloatingActionButton(
+      backgroundColor: Theme.of(context).primaryColor,
+      onPressed: () => _nextPage(controller),
+      child: const Icon(
+        Icons.chevron_right,
+        color: Colors.black,
+        size: 56 * 0.75, // 75% of fab size
+      ),
+    );
+  }
+
+  Widget _startSurveyButtonWidget(
+      BuildContext context, PageController controller) {
+    return SizedBox(
+      child: RoundedRectangleButtonWidget(
+        text: AppLocalizations.of(context)!.start_survey,
+        onPressed: () => _nextPage(controller),
+      ),
+    );
+  }
+
+  void _nextPage(PageController controller) {
+    controller.nextPage(
+      duration: const Duration(milliseconds: 300),
+      curve: Curves.easeIn,
+    );
+  }
+
+  Widget _submitSurveyButtonWidget(BuildContext context) {
+    return RoundedRectangleButtonWidget(
+      text: AppLocalizations.of(context)!.submit,
+      onPressed: () {},
+    );
+  }
+
+  bool _isIntroQuestion(QuestionUiModel question) =>
```

Comment:
What do you think about moving these properties along with `int _currentIndex = 0;` on the top? 🤔

Reply:
@markgravity They're not properties. They're functions. 🤔  Should we place them to the bottom after Widget function?

Reply:
@lydiasama my bad, I mistake with the getter function. You're good! 👍